src/mon/ConfigMonitor.cc

```diff
@@ -920,6 +920,7 @@ bool ConfigMonitor::refresh_config(MonSession *s)
 
   string device_class;
   if (s->name.is_osd()) {
+    osdmap.crush->get_full_location(s->entity_name.to_str(), &crush_location);
```

Comment:
Is the problem that the remote_host branch above doesn't fill out the host level of the location?

Reply:
Yes that's the problem because get_full_location(dev_name,...) is returning the fully qualified location of the device dev_name starting at its parent (not including dev_name).